pkg/operator/controller/ingress/controller_test.go

```diff
@@ -413,250 +485,304 @@ func TestSetDefaultPublishingStrategyHandlesUpdates(t *testing.T) {
 		name                    string
 		ic                      *operatorv1.IngressController
 		expectedIC              *operatorv1.IngressController
-		expectedResult          bool
+		platformStatus          *configv1.PlatformStatus
+		ingressConfig           *configv1.Ingress
 		domainMatchesBaseDomain bool
+		expectedResult          bool
 	}{
 		{
 			name:                    "loadbalancer scope changed from external to internal",
 			ic:                      makeIC(spec(eps(lbs(operatorv1.InternalLoadBalancer, &managedDNS))), status(eps(lbs(operatorv1.ExternalLoadBalancer, &managedDNS)))),
 			expectedResult:          true,
-			expectedIC:              makeIC(spec(eps(lbs(operatorv1.InternalLoadBalancer, &managedDNS))), status(eps(lbs(operatorv1.InternalLoadBalancer, &managedDNS)))),
+			platformStatus:          makePlatformStatus(configv1.AWSPlatformType),
+			ingressConfig:           makeIngressConfig(configv1.NLB, configv1.AWSPlatformType),
 			domainMatchesBaseDomain: true,
+			expectedIC:              makeIC(spec(eps(lbs(operatorv1.InternalLoadBalancer, &managedDNS))), status(nlbScope(operatorv1.InternalLoadBalancer, &managedDNS))),
```

Comment:
Why do you need to change the expected results of existing test cases?  This makes me nervous.